pkg/k8s/watchers/pod.go

```diff
@@ -397,7 +402,10 @@ func (k *K8sWatcher) updateK8sPodV1(oldK8sPod, newK8sPod *slim_corev1.Pod) error
 	}
 
 	for _, podEP := range podEPs {
-		if labelsChanged {
+		if labelsChanged || uidChanged {
```

Comment:
What case does this catch? That the CEP's labels are different from the Pod's labels?

If so, we should fix that regardless of the UID.

Reply:
This catches scenario 3 from the linked issue in the PR description.